sdk/identity/identity/src/msal/types.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as msalCommon from "@azure/msal-common";
+
+/**
+ * The shape we use return the token (and the expiration date).
+ * @internal
+ */
+export interface MsalToken {
+  accessToken?: string;
+  expiresOn: Date | null;
+}
+
+/**
+ * Represents the common properties of any of the MSAL responses.
+ * @internal
+ */
+export interface MsalResult {
+  account: msalCommon.AccountInfo | null;
+  accessToken: string;
+  expiresOn: Date | null;
+}
+
+/**
+ * The record to use to find the cached tokens in the cache.
+ */
+export interface AuthenticationRecord {
+  /**
+   * The associated authority, if used.
+   */
+  authority?: string;
+  /**
+   * The home account Id.
+   */
+  homeAccountId: string;
+  /**
+   * The login environment, eg "login.windows.net".
+   */
+  environment: string;
+  /**
+   * The associated tenant ID.
+   */
+  tenantId: string;
+  /**
+   * Local, tenant-specific account identifer for this account object, usually used in legacy cases.
+   */
+  localAccountId: string;
```

Comment:
This property is not in other languages.

Reply:
How do you derive this in other languages? MSAL requires us to pass this.

Reply:
What data does this contain? Local account id isn't a concept in other languages' MSAL that I've seen. Perhaps they take the this information as separate parameters, but the properties in `AuthenticationRecord` were sufficient to silently authenticate the account on subsequent execution in other languages, so I would expect this wouldn't have unique information in it and we could derive it from the fields we already have as you suggested.

Reply:
I'll ask the MSAL team.

Reply:
I was able to remove it from the public surface by re-using the homeAccountId as the same value to the localAccountId, and the tests are passing. So I'll drop localAccountId.

Reply:
homeAccountId contains the principals home tenant id, the format is `<OID>.<HomeTenantId>` I would imagine localAccountId would probably refence the tenant id the user authenticated to rather than their home tenant. So it would be the same if they authenticated to their home tenant but different for a tenant they're a guest in. If we need to give this field back to MSAL node to silently auth we should test authenticated to a guest tenant to make sure we're reconstructing it properly.

Reply:
Here's how MSAL defines it:

> MSAL first looks for the oid claim in the ID Token from an authentication response and, if present, sets it as the localAccountId in the AccountInfo object. If the oid claim is not present, MSAL falls back to setting the sub claim from the ID Token as the localAccountId.

I'm not really familiar with these, so I'm asking them on what they recommend.